resources/other/branches.md

```diff
@@ -0,0 +1,130 @@
+Working on your _own_ branch you have the safety to iterate in isolation, and only pull in your partner's features when you want them.
+
+### The Life Cycle of a Branch
+
+#### Create
+
+![Creating a branch](./images/branch-create.png)
+
+A branch typically exists tied to one feature or issue. So let's say we are working on the issue
+> \#12 Implement listDatasets
+
+Let's create a new branch named `feature/list-datasets`
+```
+$ git checkout master
+$ git pull                              # Make sure we're up to date with master
+$ git checkout -b feature/list-datasets # Create a new branch with a descriptive name
+```
+
+#### Iterate
+
+![Working on a branch](./images/branch-iterate.png)
+
+Make your changes and commit them to your branch as you did to `master` in C0, only now on `feature/list-datasets`.
+Don't forget to make [meaningful commit messages](./writing_useful_commit_messages.md)!
+
+```
+$ git add src
+$ git commit -m "#12 Implemented listDatasets stub"
+$ git push --set-upstream origin feature/list-datasets # Only done the first time
+```
+
+#### Pull Request (Iterate More)
+
+![Making a Pull Request](./images/branch-pull-request.png)
+
+Once your feature is complete, go to GitHub, click Pull Requests, and create a new pull request for your partner to [review](./code_review.md).
+
+> Implement listDatasets #17
+
+The pull request page is a space for you and your partner to discuss the changes that have been made and suggest further changes required before an approval.
+It is normal for this to require more changes! That doesn't mean your code was bad the first time. It just means it's even better with two sets of eyes on it.
+
+#### Merge
+
+![Merging a branch](./images/branch-merge.png)
+
+After you have an approval from your partner and all your tests pass, merge your pull request.
+
+> Merge pull request
+
+Done! Now don't forget to delete the branch on GitHub and on your machine.
+> Delete branch
+
+```
+$ git branch -d feature/list-datasets
+```
+
+### Branches in Concert
+
+When you have more than one branch in active development, you would expect `master` to get updated with new features while you are still working on your own branch.
+This requires one extra step for making sure your features play nice with the other ones.
+
+#### Integrate
+
+![Pulling from master](./images/branch-integrate.png)
+
+After the **Iterate** step but before the **Pull Request** step, merge the new commits on `master` into your branch.
+This gives you a chance to test your feature's integration with anything your partner (or you on another branch) has added to `master`.
+This also allows you to build, lint, and test your code _exactly_ as it will look like on `master`.
+
+```
+$ git checkout feature/list-datasets
+$ git pull origin master
+```
+Then of course commit and push the merge to `feature/list-datasets`.
```

Comment:
might be worth adding a line about: I just cloned my repo from scratch, how to I get my remote branch locally:

git checkout --track origin/<your remote branch>